spec/Section 3 -- Type System.md

```diff
@@ -2067,3 +2067,116 @@ to the relevant IETF specification.
 ```graphql example
 scalar UUID @specifiedBy(url: "https://tools.ietf.org/html/rfc4122")
 ```
+
+## Schema Coordinates
+
+SchemaCoordinate :
+  - Name
+  - Name . Name
+  - Name . Name ( Name : )
+  - @ Name
+  - @ Name ( Name : )
+
+:: A *schema coordinate* is a human readable string that uniquely identifies a
+*schema element* within a GraphQL Schema.
+
+:: A *schema element* is a specific instance of a named type, field, input
+field, enum value, field argument, directive, or directive argument.
+
+A *schema coordinate* is always unique. Each *schema element* may be referenced
+by exactly one possible schema coordinate.
+
+A *schema coordinate* may refer to either a defined or built-in *schema element*.
+For example, `String` and `@deprecated(reason:)` are both valid schema
+coordinates which refer to built-in schema elements. However it must not refer
+to a meta-field. For example, `Business.__typename` is *not* a valid
+schema coordinate.
+
+Note: A {SchemaCoordinate} is not a definition within a GraphQL {Document}, but
+a separate standalone grammar, intended to be used by tools to reference types,
+fields, and other *schema element*s. For example as references within
+documentation, or as lookup keys in usage frequency tracking.
+
+**Semantics**
+
+To refer to a *schema element*, a *schema coordinate* must be interpreted in the
+context of a GraphQL {schema}.
+
+SchemaCoordinate : Name
+  1. Let {typeName} be the value of the first {Name}.
+  2. Return the type in the {schema} named {typeName}.
+
+SchemaCoordinate : Name . Name
+  1. Let {typeName} be the value of the first {Name}.
+  2. Let {type} be the type in the {schema} named {typeName}.
+  3. If {type} is an Enum type:
+    1. Let {enumValueName} be the value of the second {Name}.
+    2. Return the enum value of {type} named {enumValueName}.
+  4. Otherwise if {type} is an Input Object type:
+    1. Let {inputFieldName} be the value of the second {Name}.
+    2. Return the input field of {type} named {inputFieldName}.
+  5. Otherwise:
+    1. Assert {type} must be an Object or Interface type.
+    2. Let {fieldName} be the value of the second {Name}.
+    3. Return the field of {type} named {fieldName}.
```

Comment:
I think this section should also specify the behaviour for Union types.
At the moment the spec does not define the behaviour for `Union.UnionTypeMember` 

```suggestion
  3. If {type} is an Union type:
    1. Assert {type} must be an Enum, Input Object, Object or Interface type.
  4. If {type} is an Enum type:
    1. Let {enumValueName} be the value of the second {Name}.
    2. Return the enum value of {type} named {enumValueName}.
  5. Otherwise if {type} is an Input Object type:
    1. Let {inputFieldName} be the value of the second {Name}.
    2. Return the input field of {type} named {inputFieldName}.
  6. Otherwise:
    1. Assert {type} must be an Object or Interface type.
    2. Let {fieldName} be the value of the second {Name}.
    3. Return the field of {type} named {fieldName}.
```

Reply:
> UnionTypeMember

The definition for this would live somewhere else in the schema, outside of the context of the union. You would write the schema coordinate to that instead.

The rule of thumb is: there should only be one way to reference a thing with Schema Coordinates

Reply:
@magicmark  I understand, though the algorithm is still wrong.

Reply:
@magicmark  Makes sense to me

Reply:
@leebyron is it worth keeping in the clarification (or something similar) from a previous commit?

<img width="617" alt="Screen Shot 2021-12-31 at 4 42 48 AM" src="https://user-images.githubusercontent.com/1590756/147824058-8addb64b-76ef-4c82-b976-0dca94a96f44.png">

My gut says this may crop up again and cause some confusion - worth a counter example?

Reply:
> @leebyron is it worth keeping in the clarification (or something similar) from a previous commit?
> 
> [magicmark/graphql-spec@`ac5318f`/spec/Section%203%20--%20Type%20System.md?plain=1#L391-L399](https://github.com/magicmark/graphql-spec/blob/ac5318f9a9f04d0f8dd03d3758d5ca2654f7f7d7/spec/Section%203%20--%20Type%20System.md?plain=1#L391-L399)
> 
> My gut says this may crop up again and cause some confusion - worth the counter example?

Just chiming in. From my perspective the counter example would be ideal (maybe even clarify that directives on types are also invalid?).

Maybe add a line on *why* union members are not valid schema coordinates, instead of solely stating that they are not?

```diff
- Note: You may not select members inside a union definition.
+ Note: Union members do not uniquely identify elements in the schema. Therefore the following counter example is *not* considered a valid Schema Coordinate:
```

...or something along those lines

Reply:
fwiw this is all sort of encapsulated by the following line the spec:

> A *schema coordinate* is always unique. Each *schema element* may be referenced
> by exactly one possible schema coordinate.

++ for the explainer, I would maybe try and reuse this language, or hyperlink back to this source of truth somehow